src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -82,23 +86,35 @@ impl WorkUnit {
   }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+enum WorkunitRecord {
+  Started(StartedWorkUnit),
+  Completed(WorkUnit),
+}
```

Comment:
Hm, this name makes me think that a `Workunit` should be the combination of the StartedData and CompletedData. You know how this gets used much better than me, though. Does that suggestion make sense?